sdk/helper/ldaputil/client.go

```diff
@@ -31,6 +31,12 @@ func (c *Client) DialLDAP(cfg *ConfigEntry) (Connection, error) {
 	var retErr *multierror.Error
 	var conn Connection
 	urls := strings.Split(cfg.Url, ",")
+
+	// Default timeout in the pacakge is 60 seconds, which we default to on our
+	// end. This is useful if you want to take advantage of the URL list to increase
+	// availability of LDAP.
+	ldap.DefaultTimeout = time.Duration(cfg.ConnectionTimeout) * time.Second
```

Comment:
Instead of changing the external package's `DefaultTmeout`, there is an `ldap.DialURL` that allows for a `DialWithDialer` option where we can set the timeout.

Reply:
I looked into this but I think it will require a lot more changes to this helper library instead of modifying the Dial wrapper. That seemed a lot more risky even though I don't take changing an exported package variable lightly.

Reply:
@jasonodonnell it looks like this is triggering the race detector in tests, since two tests running in parallel modify `ldap.DefaultTimeout`. https://github.com/hashicorp/vault/actions/runs/4697499717/jobs/8329395640. I can remove the parallel tests as a workaround, but this is perhaps more reason to switch to `DialURL`